app/src/main/java/chat/rocket/android/authentication/domain/model/DeepLinkInfo.kt

```diff
@@ -0,0 +1,99 @@
+    val userId: String?,
+    val token: String?,
+    val rid: String?,
+    val roomType: String?,
+    val roomName: String?
+) : Parcelable
+
+fun Uri.getDeepLinkInfo(): DeepLinkInfo? {
+    return if (isAuthenticationDeepLink()) {
+        val host = getQueryParameter("host")
+        val url = if (host.startsWith("http")) host else "https://$host"
+        val userId = getQueryParameter("userId")
+        val token = getQueryParameter("token")
+        try {
+            DeepLinkInfo(url, userId, token, null, null, null)
+        } catch (ex: Exception) {
+            Timber.d(ex, "Error parsing auth deeplink")
+            null
+        }
+    } else if (isCustomSchemeRoomLink()) {
+        val hostValue = getQueryParameter("host")
+        val url = if (hostValue.startsWith("http")) hostValue else "https://$hostValue"
+        val rid = getQueryParameter("rid")
+        val pathValue = getQueryParameter("path")
+        val pathSplit = pathValue.split("/")
+        val roomType = pathSplit[0]
+        val roomName = pathSplit[1]
+        try {
+            DeepLinkInfo(url, null, null, rid, roomType, roomName)
+        } catch (ex: Exception) {
+            Timber.d(ex, "Error parsing custom scheme room link")
+            null
+        }
+    } else if (isWebSchemeRoomLink()) {
+        val url = "https://$host"
+        val pathSplit = path.split("/")
+        val roomType = pathSplit[1]
+        val roomName = pathSplit[2]
+        try {
+            DeepLinkInfo(url, null, null, null, roomType, roomName)
+        } catch (ex: Exception) {
+            Timber.d(ex, "Error parsing login deeplink")
+            null
+        }
+    } else null
+}
+
+fun Intent.isSupportedLink(): Boolean {
+    return (action == Intent.ACTION_VIEW && data != null &&
+            (data.isDynamicLink() || data.isAuthenticationDeepLink() ||
+                    data.isCustomSchemeRoomLink() || data.isWebSchemeRoomLink()))
+}
+
+fun Uri.isDynamicLink(): Boolean {
+    return (host != null && host.contains("page.link", ignoreCase = true))
+}
+
+// Authentication deep link defined here: https://rocket.chat/docs/developer-guides/deeplink/#authentication
+private inline fun Uri.isAuthenticationDeepLink(): Boolean {
+    if (host == "auth")
+        return true
+    else if (host == DEFAULT_SERVER_HOST && path == "/auth")
+        return true
+    return false
+}
+
+// Custom scheme room deep link defined here: https://rocket.chat/docs/developer-guides/deeplink/#channel--group--dm
+private inline fun Uri.isCustomSchemeRoomLink(): Boolean {
+    if (scheme.startsWith("rocketchat") &&
+            host == "room")
+        return true
+    return false
+}
+
+// http(s) scheme deep link not yet documented. Ex: https://open.rocket.chat/direct/testuser1
+private inline fun Uri.isWebSchemeRoomLink(): Boolean {
+    val roomType = path.split("/")[1]
+    if (scheme.startsWith("http") &&
+            (roomType == "channel" || roomType == "group" || roomType == "direct"))
```

Comment:
You can make use of `roomTypeOf(roomType)` here to check is the room type is channel, group, etc.

Ex.:

`... roomTypeOf(roomType) is RoomType.DirectMessage`

Reply:
@filipedelimabrito will roomTypeOf() just accept the lower case strings we've parsed from the path?  Or does it need to be converted into some object or capitalized string?  It's not clear to me right away.  Thanks.